### PR TITLE
StructuredRecordToPutTransformer, StructuredRecordToAvro Transformer

### PR DESCRIPTION
**Pull request by gokulavasan**
Takes in an output schema and filters out fields that are not present in it.

This allows the sink to drop fields if the schema string that it gets has fields removed compared to the previous stage. Before this change, we were converting the structured record we get directly to Put or AvroRecord without taking into consideration the output schema that was provided (output schema was used only for explore). 

Build : http://builds.cask.co/browse/CDAP-RBT413-1

JIRA :

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/TableSink.java [hunk 1]
```diff
@@ -61,7 +62,14 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   @Override
   public void initialize(BatchSinkContext context) throws Exception {
     super.initialize(context);
-    recordPutTransformer = new RecordPutTransformer(tableSinkConfig.getRowField(),
+    Schema outputSchema = null;
```
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/TableSink.java]: What happens when you don't specify any output schema? Will everything be filtered out? Can we add a test case for that?

Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/TableSink.java]: If outputSchema is null, all the fields from the StructuredRecord are 'put' into Put. This was the behavior earlier. The old tests already test for this. The test that I have added also tests again for the case of outputSchema being null.

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/template/etl/common/RecordPutTransformerTest.java [hunk 2]
```diff
@@ -121,11 +121,31 @@ public void testCaseInsensitiveRowKeyValid() {
...
+                                         Schema.Field.of("age", Schema.of(Schema.Type.INT)));
+    StructuredRecord record = StructuredRecord.builder(inputSchema)
+      .set("id", 123L).set("name", "ABC").set("age", 10).build();
+    RecordPutTransformer transformer = new RecordPutTransformer("id", outputSchema, true);
+    Put put = transformer.toPut(record);
+    Assert.assertEquals(1, put.getValues().size());
+
+    transformer = new RecordPutTransformer("id", null, true);
+    put = transformer.toPut(record);
+    Assert.assertEquals(2, put.getValues().size());
```
Comment (gsps1) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/template/etl/common/RecordPutTransformerTest.java]: can you add one more output schema without any fields?

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/TableSink.java (hunk 1 above)
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/TableSink.java]: ok

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/template/etl/common/RecordPutTransformerTest.java (hunk 2 above)
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/template/etl/common/RecordPutTransformerTest.java]: let's add this test to develop/

**Comment by sreevatsanraman** [comment]
LGTM :+1:
